src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/ConstVal.cs

```diff
@@ -20,149 +21,77 @@ internal enum ConstValKind
         Decimal,
         IntPtr,
         Float,
-        Boolean,
-        Lim
-    };
-
+        Boolean
+    }
 
-    internal sealed class CONSTVAL
+    internal sealed class ConstVal
```

Comment:
That's a good idea, though perhaps still worth caching the boxed values. I'll look into that.